src/core/services/ws.ts

```diff
@@ -254,14 +254,21 @@ export class CoreWSProvider {
             // Create the tmp file as an empty file.
             const fileEntry = await CoreFile.createFile(tmpPath);
 
-            const transfer = FileTransfer.create();
-            onProgress && transfer.onProgress(onProgress);
+            const transfer = new window.FileTransfer();
+
+            onProgress && transfer.onprogress(onProgress[1]);
```

Comment:
```suggestion
            if (onProgress) {
                transfer.onprogress = onProgress;
            }
```
This line is wrong, using onProgress[1] in a callback will return undefined IMO. The reason TS was complaining is because onprogress needs to be assigned like this.